main_no_metrics_test.go

```diff
@@ -0,0 +1,1003 @@
+package main_test
```

Comment:
Same set of tests, as we have for e2e, but without duplicated ones in case if we remove metrics. Some of those becoming same, if we remove metrics check

Reply:
maybe let's try to avoid copy-pasting with such approach:

```go
func TestE2E...(t *testing.T) {
    test logic ...
    
    if SKIP_METRICS { return }

    metrics logic...
}
```

Reply:
Refactored